### PR TITLE
recoll: fix darwin build

### LINKED ISSUE #272594: Build failure: recoll
### Steps To Reproduce

Note: `catdvi` is a recoll dependency that previously didn't build (#272384); @a-n-n-a-l-e-e graciously fixed it but it currently isn't merged. This reflects their fix:
1. `nix profile install github:NixOS/nixpkgs/9559cea14a91ffa63f5f90d7719b2c56233517f2#recoll`

(doesn't build on macOS; builds fine elsewhere)

### Build log

`nix log /nix/store/kn6a1vj3374s3vq56srkskdclwjzaci5-recoll-1.36.2.drv`
```
https://gist.github.com/aidan-gibson/c4f3f580186fe515d0dfd2f75d8a818b
```
tl;dr is 
```
> Project ERROR: Unknown module(s) in QT: webenginewidgets
> make[2]: *** [Makefile:3014: recollqt] Error 3
```

### Metadata

```console
nix-shell -p nix-info --run "nix-info -m"
these 3 paths will be fetched (1.02 MiB download, 8.17 MiB unpacked):
  /nix/store/ww2aj2qyn7mpd6pizy8ik22bdpzivf03-DarwinTools-1
  /nix/store/xhvxpfkb3xmicav5wdx9xfyiaw452mbl-bash-interactive-5.2-p15
  /nix/store/ngbvqkdnz9yaxmf9qc1lpyr05ya9g1k4-nix-info
copying path '/nix/store/ww2aj2qyn7mpd6pizy8ik22bdpzivf03-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/xhvxpfkb3xmicav5wdx9xfyiaw452mbl-bash-interactive-5.2-p15' from 'https://cache.nixos.org'...
copying path '/nix/store/ngbvqkdnz9yaxmf9qc1lpyr05ya9g1k4-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/wk5dq7iwpd8yb4mnhb98k687kgdqachh-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
possibly fixes: #272594
builds and creates a gui on darwin, not sure how much functionality works or doesn't.

the executables in bin were being double wrapped, explicitly by wrapProgram, then implicitly by the qt Framework. just add the needed flags to `qtWrapperArgs` and removed the wrapProgram from the postInstall script. verified single wrap and have correct env vars set.

Result of `nixpkgs-review pr 272854` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.recoll</li>
    <li>recoll (python311Packages.recoll)</li>
  </ul>
</details>

<hr>

Result of `nixpkgs-review pr 272854` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.recoll</li>
    <li>recoll (python311Packages.recoll)</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/search/recoll/0001-no-qtgui-darwin-bundle.patch [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+diff --git a/qtgui/recoll.pro.in b/qtgui/recoll.pro.in
+index a0ef314..42d4633 100644
+--- a/qtgui/recoll.pro.in
++++ b/qtgui/recoll.pro.in
+@@ -180,202 +180,7 @@ windows {
+   }
+ }
+ 
+-macx: {
```
Comment (marsam) [inline pkgs/applications/search/recoll/0001-no-qtgui-darwin-bundle.patch:18]: I think we can make this patch smaller with
```diff
 }
 
-macx: {
+if(false) {
   QCBUILDLOC=Qt_6_4_2_for_macOS
```

**Comment by marsam** [comment]
Looks great, thanks! I tested on x86_64-darwin
Can we remove the draft status?

**Comment by a-n-n-a-l-e-e** [comment]
@ofborg build recoll recoll.passthru.tests

**Comment by a-n-n-a-l-e-e** [comment]
i should rebase this to get the updated version.

**Review by marsam — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**